src/resources/uart_esp32.cc

```diff
@@ -461,10 +462,14 @@ UART_ISR_INLINE void UartResource::disable_tx_interrupts(bool done) {
   }
 }
 
+UART_ISR_INLINE bool UartResource::has_flushed_tx_fifo() {
+  return get_tx_fifo_free() == SOC_UART_FIFO_LEN && uart_toit_hal_is_tx_idle(hal_);
```

Comment:
Maybe add a comment that explains why we need to check for idle? It wasn't clear to the original writer of the code :)

Reply:
done.